### DISCUSSION TITLE
Custom axios instance for each endpoint?

### DISCUSSION THREAD

**Discussion by patjakubik** [ecyrbe/zodios#423] [Q&A]
Greetings,

I am currently working with an API that seems to have a tendency towards timeouts on some of its endpoints. To address this, I have set up a custom axios instance with interceptors that handle retry logic. However, I am encountering an issue where some endpoints require a timeout of 2-3 seconds, while others require up to 30 seconds.

The problem lies in the fact that the Axios instance is attached globally to a group of endpoints within Zodios. As a result, it seems that the only way to work around this would be to create multiple Zodios instances for every endpoint that requires a different axios instance. Unfortunately, this solution doesn't feel optimal.

Do you have any suggestions for a better approach to this issue?

**Comment by ecyrbe** [comment]
You can attach zodios plugins per endpoint. Zodios plugins are like axios interceptors but with knowledge about api description.
See docs about plugins :
https://www.zodios.org/docs/client/plugins

**Reply by patjakubik** [reply]
Could example here https://www.zodios.org/docs/client/plugins#write-your-own-plugin include imports?

I believe that `ZodiosEnpointDescriptions` is a typo, should it be `ZodiosEndpointDefinitions`? Also how about `Error`? Is this supposed to be generic Node Error?

**Reply by ecyrbe** [reply]
Yes, it's a typo.
Error is correct, you'll need to check if it's a axios error or a generic error if you want to handle error.

**Reply by ecyrbe** [reply]
You can check zod validation internal plugin as an example
